lbrynet/core/HTTPBlobDownloader.py

```diff
@@ -0,0 +1,81 @@
+from random import choice
+import logging
+
+from twisted.internet import defer
+import treq
+from twisted.internet.task import LoopingCall
+
+log = logging.getLogger(__name__)
+
+
+class HTTPBlobDownloader(object):
+    def __init__(self, blob_manager, blob_hashes=None, servers=None, client=None):
+        self.blob_manager = blob_manager
+        self.servers = servers or []
+        self.client = client or treq
+        self.blob_hashes = blob_hashes or []
+        self.looping_call = LoopingCall(self._download_next_blob_hash_for_file)
+        self.failures = 0
+        self.max_failures = 3
+        self.interval = 1
+
+    @property
+    def running(self):
+        return self.looping_call.running
+
+    def start(self):
+        if not self.running and self.blob_hashes and self.servers:
+            return self.looping_call.start(self.interval, now=True)
```

Comment:
Shouldn't this return `defer.succeed(True)` in the implicit `else` case? Otherwise it requires the caller to `defer.maybeDeferred()` your `start()` method. I think it's good etiquette to keep an `async` method `async` in all circumstances.

Reply:
Fixed, thanks. (the comment isn't gone because it's the other line)